workstation-backend/workspace/views.py

```diff
@@ -1,3 +1,17 @@
-from django.shortcuts import render
+from django.db.utils import IntegrityError
+from rest_framework import generics, mixins, serializers
 
-# Create your views here.
+from .models import Background
+from .serializers import BackgroundSerializer
+
+
+class BackgroundView(mixins.CreateModelMixin, generics.RetrieveUpdateAPIView):
+    lookup_field = "workspace_id"
+    queryset = Background.objects.all()
+    serializer_class = BackgroundSerializer
+
+    def post(self, request, *args, **kwargs):
```

Comment:
I think this should be part of the `BackgroundSerializer`
Check out:
https://www.django-rest-framework.org/tutorial/1-serialization/

Reply:
For this, I'm following: https://www.django-rest-framework.org/tutorial/3-class-based-views/#using-mixins